client/src/main/java/io/pravega/schemaregistry/client/SchemaRegistryClient.java

```diff
@@ -0,0 +1,364 @@
+     * or {@link SchemaRegistryClient#getGroupHistory} or {@link SchemaRegistryClient#getSchemas} or 
+     * {@link SchemaRegistryClient#getLatestSchemaVersion}
+     * If add schema is called again using this deleted schema will result in a new version being assigned to it upon registration. 
+     * 
+     * @param groupId Id for the group. 
+     * @param versionInfo Version which uniquely identifies schema within a group. 
+     * @throws ResourceNotFoundException if group is not found. 
+     * @throws UnauthorizedException if the user is unauthorized.
+     */
+    void deleteSchemaVersion(String groupId, VersionInfo versionInfo) throws ResourceNotFoundException, UnauthorizedException;
+
+    /**
+     * Deletes the schema associated to the given version. Users should be very careful while using this API in production, 
+     * esp if the schema has already been used to write the data. 
+     * An implementation of the delete call is expected to be idempotent. The behaviour of delete schema API invocation 
+     * with the schema registry service is idempotent. 
+     * The service performs a soft delete of the schema. So getSchemaVersion with the version info will still return the schema.
+     * However, the schema will not participate in any compatibility checks once deleted.
+     * It will not be included in listing schema versions for the group using APIs like {@link SchemaRegistryClient#getSchemaVersions}
+     * or {@link SchemaRegistryClient#getGroupHistory} or {@link SchemaRegistryClient#getSchemas} or 
+     * {@link SchemaRegistryClient#getLatestSchemaVersion}
+     * If add schema is called again using this deleted schema will result in a new version being assigned to it upon registration. 
+     * 
+     * @param groupId Id for the group.
+     * @param schemaType schemaType that identifies the type of object the schema represents. This should be same as the 
+     *                   value specified in {@link SchemaInfo#type}. 
+     * @param version Version number which uniquely identifies schema for the schemaType within a group. 
+     * @throws ResourceNotFoundException if group is not found. 
+     * @throws UnauthorizedException if the user is unauthorized.
+     */
+    void deleteSchemaVersion(String groupId, String schemaType, int version) throws ResourceNotFoundException, UnauthorizedException;
+
+    /**
+     * Gets schema corresponding to the version. 
+     * 
+     * @param groupId Id for the group. 
+     * @param versionInfo Version which uniquely identifies schema within a group. 
+     * @return Schema info corresponding to the version info.
+     * @throws ResourceNotFoundException if group or version is not found. 
+     * @throws UnauthorizedException if the user is unauthorized.
+     */
+    SchemaInfo getSchemaForVersion(String groupId, VersionInfo versionInfo) throws ResourceNotFoundException, UnauthorizedException;
```

Comment:
I think two comment threads did not get ported over to this PR. I was asking about the possibility of this method on and the one below returning an `EncodingInfo`

Reply:
This api is not related to encoding id. 
This method returns the Schema corresponding to a version info object. 

An encoding id uniquely identifies {schema version + codecType} pair. 
And there is an api to get encoding id for that pair called getEncodingId.

Reply:
That does not answer my question. The only difference between `SchemaInfo` and `EncodingInfo` is the additional field `codecType`. Does this mean that given a groupId and a versionInfo it is not possible to lookup the `codecType`? Are there multiple codecTypes for a versionInfo?

Reply:
> Are there multiple codecTypes for a versionInfo?
A group registers two types of entities -- schemas and codecTypes. 
And on demand it generates a unique encoding id identifying each combination of the two. 
So if you have 10 schema versions and 2 codecTypes registered with the service, then 20 encoding ids could be generated by the service. 

> Does this mean that given a groupId and a versionInfo it is not possible to lookup the codecType

There are apis to add codecType and getAllCodecTypes that the service provides for a group.

Reply:
I think we should more tightly integrate these concepts. For example, I don't think we want to imply that adding a new schema should only check the schema validation rules and it does not matter what encoding the data is sent with.

Instead we could have encodings as a property of the schema version. This way encodings and codecs can be included in compatibility checks, and any future Rules validation logic.

This would mean that codec would become a field in SchemaInfo. This would allow us to simplify this top level API.

Reply:
Encoding is not part of schema definition. Take the example of compression.

Compressing data seralized with Schema version 0 is not the property of schema describing the structure of the data. 
It is an encoding applied on top of serialized blob. 
Similarly, encryption codec could be used to encrypt the data. That isnt a property of the structure of the data and should not be included there as such. 

Keeping it decoupled allows us to not declare at the time of schema addition which all encodings could be used on the schema at the write time. 

And application writers can choose independently if they want to perform any encoding while writing the data with whichever schema they want to write data with.

Reply:
This would mean if an application is reading and doesn't support gziped data, a writer could start gzipping what it is writing, and the service will allow it causing readers to break at Runtime.

Reply:
So adding a codec is not allowed based on the policy? That should be documented.

Reply:
The javadoc on addCodecType method describes that add codec is always backward incompatible. 

Presently there is no policy governance for addCodec. 

Adding a new Codec is always Backward incompatible and readers need to be upgraded first whenever a new codec is added. To elaborate: unlike schema compatibility where comparisons with different schemas need to be made in different direction, there is no such criteria here. 

So its either allowed or disallowed, and when allowed its always backward incompatible. 

So as a policy we can add a flag to group properties with an API to toggle enable-addCodecType.
I have created an issue to evaluate policy options governing codecTypes.
https://github.com/pravega/schema-registry/issues/39